### PR TITLE
MUMUP-2334 : Silent login on load

### PR DESCRIPTION
**Pull request by timlevett**
+ During loading if a recent login is not detected (last 4h) then it does a silent login via `/portal/Login?silent=true`. 
+ The response is JSON, which we store in the `$sessionStorage.portal` object. username, lastLoginMills, and theme.
+ During a fetch to `session.json` it gets a different username that is stored in `$sessionStorage.portal.username` it will throw a redirect to the login page
+ By default the silent login on load is disabled (as most apps won't do this, but `angularjs-portal` will)
+ Documented all new `app-config.js` entries in configuration.md. Lucky enough, if you don't want to use this feature, then just don't configure any of the new entries.

### CODE REVIEW

**Comment by apetro** [comment]
> Lucky enough, if you don't want to use this feature, then just don't configure any of the new entries.

:heart:

File: uw-frame-components/portal/main/services.js [hunk 1]
```diff
@@ -15,6 +15,16 @@ define(['angular'], function(angular) {
...
       userPromise = prom.then(
           function(data, status) { //success function
+            if(APP_FLAGS.loginOnLoad) {
+              //quick check to make sure you are who your browser says you are
+              if($sessionStorage.portal
+                && $sessionStorage.portal.username
+                && data.data.person.userName != $sessionStorage.portal.username) {
+                  console.warn("Thought they were " + $sessionStorage.portal.username +
+                   " but session sent back " + data.data.person.userName +". Redirect!");
+                  miscService.redirectUser(302, "Wrong User than populated in session storage.");
```
Comment (apetro) [inline uw-frame-components/portal/main/services.js]: Is this redirect sufficient cleanup, or should more be done here to clear sessionStorage?

Comment (timlevett) [inline uw-frame-components/portal/main/services.js]: True, could delete `$sessionStorage.portal` to really get it clean

Comment (timlevett) [inline uw-frame-components/portal/main/services.js]: Doing.

**Comment by vertein** [comment]
:+1:

**Comment by paulerickson** [comment]
Sounds like a new, backwards-compatible feature, so a minor version bump might be in order?

**Comment by timlevett** [comment]
Logical. will do @paulerickson

**Pull request merged**